### ISSUE TITLE
Nextcloud compatibility?

### ISSUE THREAD

**Issue by poVoq** [jobisoft/TbSync#115]
## Your environment

TbSync version: 0.7.12
Thunderbird version: 60.0

[ x ] Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

## Expected behavior
Can establish connection with Nextcloud CalDAV/CardDAV service

## Actual behavior
Setup fails.

## Steps to reproduce
Install and try to use with Nextcloud 13.0.5

## Other remarks
Might be related to this bug: 
https://help.nextcloud.com/t/thunderbird-60-problems-with-address-and-calendar-sync/35773
With the explained work around CalDAV syncing with default Thunderbird works.

**Comment by jobisoft** [comment]
I have confirtmation, that it works with Nextcloud. Just to make sure, please update to the latest version avail:
https://github.com/jobisoft/DAV-4-TbSync/releases/download/v0.8/DAV-4-TbSync.xpi
https://github.com/jobisoft/TbSync/releases/download/v0.7.12/TbSync-beta.xpi

Also please enable debug log and send me the log after teh setup attempt failed.

**Comment by poVoq** [comment]
Ok seems to work with the DAV-4-TBSync.xpi you linked. I had the one from the Mozilla store and it failed.
Thanks!

**Comment by jobisoft** [comment]
Yeah, mozilla is taking long for review at the moment. The version you installed just now will update itself via github ("self-hosted"). I will later switch you back over to the mozilla store...

**Comment by poVoq** [comment]
Hmm, any idea why the calendars are deactivated and when enabling them they get disabled directly again?
CardDAV seems to work though and I get no sync errors.

**Comment by jobisoft** [comment]
That is the issue you pointed out in your remark. TbSync has its own CardDav implementation, which is not prone to this error. The CalDAV sync however is done by Lightning, TbSync just adds the calendars. 

If this does not work even with the setting switched to false, you have to disable calendar sync in TbSync and set up the calendars by hand in lightning (just as you did before before). But I would not understand, why this would be necessary.

**Comment by poVoq** [comment]
That is odd... I have the setting to false and it works by manually adding the calendars, but not through TbSync it seems.

**Comment by jobisoft** [comment]
It works with ownCloud. I do not have a Nextcloud but I heard that before. Any chance I could get a test account on your server?

**Comment by poVoq** [comment]
Send you an email with login details for a test account. Thanks a lot for testing it!

**Comment by jobisoft** [comment]
Thanks, I will try to find out whats causing this.

**Comment by poVoq** [comment]
This seem to be the relevant output of the Lightning debug log when I try to enable a calendar added through TBsync.
```
Lightning:CalDAV: send: <?xml version="1.0" encoding="UTF-8"?>
<D:propfind xmlns:D="DAV:" xmlns:CS="http://calendarserver.org/ns/" xmlns:C="urn:ietf:params:xml:ns:caldav"><D:prop><D:resourcetype/><D:owner/><D:current-user-principal/><D:supported-report-set/><C:supported-calendar-component-set/><CS:getctag/></D:prop></D:propfind>  calDavCalendar.js:1699

Lightning:CalDAV: Status 404 on initial PROPFIND for calendar TestCalendar  calDavCalendar.js:1706
```
Since you were able to connect correctly to my German server from Germany, I suspect that this might be related to it requesting this additional external resource that might be blocked in the country I am trying to access this from right now (Bangladesh)?

**Comment by jobisoft** [comment]
Could you send me a larger section of the log, more lines before and after, so I can see "where" this happens during the handshake? I would need some URLs, request and response payloads before and after. If confidential, send it via email.

**Comment by jobisoft** [comment]
Is TbSync using the same URL to setup the calendar, as you do by hand?

**Comment by jobisoft** [comment]
I added, to actually enforce a sync of the calender (a "refresh") after creating it. That did the trick? That would be nice. If the error should re-appear, just re-open this issue.

**Comment by poVoq** [comment]
ok I played around with it more and also tested it on my other Win10 machine where the issue originally surfaced. Looks like this error isn't fixed after all, but not related to Nextcloud per se.

Even after installing the latest version on the Win10 PC it still doesn't work as before. The installation history on that machine was: Fresh install of Thunderbird 60, manual addition of individual Nextcloud calendar. Discovery of TBSync, Removal of manually added calendar and attempt to configure through Tbsync, Not working, so added calendar manually again, which works.

On the TB60 Linux PC I tried just now the following happend. Never had CalDAV sync configured. Added directly TBSync and configured Nextcloud. It works after that small config addition linked in the first post (yay). Then I tried adding another CalDAV service (HubZilla, but that also uses SabreDAV like Nextcloud) and again the same error with the calender turning directly off again also happens on the Linux machine.

So it seems to be that the very first CalDAV server added works, but if you have of have had ever any CalDAV configured before (either manually or through TBSync) then it fails for the next one.
Does that even make any sense?

**Comment by jobisoft** [comment]
In the meantime, could you "restart" on any of your machines (new profile) and set up the two accounts in swapped order? First HubZilla and than Nextcloud? Would be interesting to know, if this is consistent and HubZilla will work, but Nextcloud will fail...

**Comment by jobisoft** [comment]
I am not able to break my system, I cannot reproduce this at all:

![image](https://user-images.githubusercontent.com/5830621/44868427-1191ce80-ac8b-11e8-823a-b6edc6a66cde.png)

That are 4 different servers, nextcloud being yours. I can disable and re-enable them as I want. It just works. Bugs I cannot reproduce are the worst.

Can you disable all other Addon, maybe it is a conflict with some other AddOn?

**Comment by poVoq** [comment]
No other addons installed.

Ok lets files this under extremely wired and move on for now. After all the Nextcloud works :smile:

**Comment by jobisoft** [comment]
If you could send me the lightning log of a failed attempt, that might give me some details.

**Comment by jobisoft** [comment]
It happens on both your systems?

**Comment by poVoq** [comment]
Like I wrote HubZilla fails on both systems now. Nextcloud used to fail on one, but after completly resetting it and installing the latest version of your plugins it seem to work on both.

**Comment by poVoq** [comment]
In the lightning log there is nothing elserelated than what I posted above. Of course there is now all the stuff from the working Nextcloud CalDav in addition, but the HubZilla releated entries are exactly the same as before with Nextcloud on the Win10 PC.

**Comment by jobisoft** [comment]
Thanks for your help. I can totally understand, if you do not want to spend more time on this. If you have any new findings, just let me now.

One last thing: The 404 Error you get in the lightning log, is he using the correct URL? Does the URL look wrong in any way? Do you get a 404 on that URL, if you paste it into your browser?

**Comment by poVoq** [comment]
Don't get me wrong, I really appreciate the time and effort you put into trying to find this bug with me. I just don't think we will progress much further with this trial an error for now. Of course if I come across another hint I'll share it immediately. But like the wrong hint I shared above, I also don't want to cause unnecessary effort on your side by my mostly unfounded speculation :(

The "https://www.calendarserver.org/ns/" URL pasted in my browser does give a 
```
Calendar and Contact Server
Not Found 
The requested URL was not found on this server.
```
page here, which is likely a custom 404 page?

**Comment by jobisoft** [comment]
Oh no, that specific log entry does not include the URL. That sucks, impossible to debug.

Other question: Are we talking about calendars whose name includes "special" chars? German Umlauts?

**Comment by poVoq** [comment]
With 
```
calendar.debug.log
calendar.debug.log.verbose
```
enabled and the working Nextcloud CalDAV disabled in Tbsync, this is all I get in the Error Console from Thunderbird:
```
Lightning:[calSleepMonitor] Starting sleep monitor.  calSleepMonitor.js:60
While creating services from category 'profile-after-change', could not create service for entry 'calendar-backend-loader', contract ID 'service,@mozilla.org/calendar/backend-loader;1'
Lightning:"[calTimezoneService] Loading resource://calendar/timezones/zones.json"  calTimezoneService.js:71
Lightning:[calTimezoneService] Timezones version 2.2018e loaded  calTimezoneService.js:122
Use of Mutation Events is deprecated. Use MutationObserver instead.  calendar-widgets.xml:512:20
Lightning:CalDAV: send: <?xml version="1.0" encoding="UTF-8"?>
<D:propfind xmlns:D="DAV:" xmlns:CS="http://calendarserver.org/ns/" xmlns:C="urn:ietf:params:xml:ns:caldav"><D:prop><D:resourcetype/><D:owner/><D:current-user-principal/><D:supported-report-set/><C:supported-calendar-component-set/><CS:getctag/></D:prop></D:propfind>  calDavCalendar.js:1699
Lightning:CalDAV: Status 404 on initial PROPFIND for calendar HubzillaCalendar  calDavCalendar.js:1706
Lightning:CalDAV: send: <?xml version="1.0" encoding="UTF-8"?>
<D:propfind xmlns:D="DAV:" xmlns:CS="http://calendarserver.org/ns/" xmlns:C="urn:ietf:params:xml:ns:caldav"><D:prop><D:resourcetype/><D:owner/><D:current-user-principal/><D:supported-report-set/><C:supported-calendar-component-set/><CS:getctag/></D:prop></D:propfind>  calDavCalendar.js:1699
Lightning:CalDAV: Status 404 on initial PROPFIND for calendar HubzillaCalendar  calDavCalendar.js:1706
Lightning:CalDAV: send: <?xml version="1.0" encoding="UTF-8"?>
<D:propfind xmlns:D="DAV:" xmlns:CS="http://calendarserver.org/ns/" xmlns:C="urn:ietf:params:xml:ns:caldav"><D:prop><D:resourcetype/><D:owner/><D:current-user-principal/><D:supported-report-set/><C:supported-calendar-component-set/><CS:getctag/></D:prop></D:propfind>  calDavCalendar.js:1699
Lightning:CalDAV: Status 404 on initial PROPFIND for calendar HubzillaCalendar  calDavCalendar.js:1706
```
I tried to enable the calendar 3 times manually.

No Umlaute, sorry.

**Comment by jobisoft** [comment]
I let you go in just 1 minute :-) 

Can you send me the URLs of the non-working calendars you find in their "edit" dialog? Including any leading/trailing spaces...

![image](https://user-images.githubusercontent.com/5830621/44873080-8965f600-ac97-11e8-94c9-abbb691ed8db.png)

If you tick the enable box, you can select and copy the entry.

**Comment by poVoq** [comment]
Sure I can send you by email, but manually adding them works in TB.

**Comment by jobisoft** [comment]
Yes, I know, but I need to see the exact (!) URLs that TbSync is adding.

**Comment by plcstop** [comment]
I had this same issue. It appeared in exactly the same behaviour not matter if using TBsync or Lightning directly.
I could solve it by deactivating all calenders. Then removed ALL stored passwords related to my server. Then activated the calendards again (and re-entering the passwords).
That's all

**Comment by monperrus** [comment]
FYI: cannot connect to self-hosted Nextcloud with TB 60.8.0 and TbSync 1.7.0.2

**Comment by jobisoft** [comment]
That issue has been solved long ago. TbSync works perfectly with nextcloud. Use auto setup (discovery) and just provide the base URL of your server.

**Comment by monperrus** [comment]
Auto-setup does not work, probably because `.well-known` urls are not well handled by the reverse proxy.

However, by trying again, I found the problem: the URL is case-sensitive, and the address book `Contacts` has an internal identifier `contacts`, so

```
// works
https://nextcloud.myserver.net/remote.php/dav/addressbooks/users/admin/contacts

// fails
https://nextcloud.myserver.net/remote.php/dav/addressbooks/users/admin/Contacts
```

**Comment by jobisoft** [comment]
Glad it works now, but just a side note: You do not need to specify the actual resource. Even if auto/discovery setup does not work due to the issue with the `.well-known` urls, you can still use resource discovery.

What happens, if you just use `https://nextcloud.myserver.net/remote.php/dav/` as caldav and cardav url in manual setup?

If that does not work, what about
* `https://nextcloud.myserver.net/remote.php/caldav/`
* `https://nextcloud.myserver.net/remote.php/carddav/`

for the corresponding server urls?

**Comment by monperrus** [comment]
> What happens, if you just use https://nextcloud.myserver.net/remote.php/dav/ as caldav and cardav url in manual setup?

it also works!

**Comment by elastic2ls-awiechert** [comment]
Hi,
neither of the url combinations I tried will work. I also checked autodiscovery funtion without luck. But it works for calenders in Lighting extension and calendars/contact on iphone.


    https://nextcloud.myserver.net/remote.php/caldav/
    https://nextcloud.myserver.net/remote.php/carddav/
    https://nextcloud.myserver.net/remote.php/dav

`URL:
https://nextcloud.myserver.net/remote.php/dav (PROPFIND)

Request:
<d:propfind xmlns:d="DAV:"><d:prop><d:current-user-principal /></d:prop></d:propfind>

Response:`

**Comment by jobisoft** [comment]
please create a new issue with all the requiered info about your system.
